packages/client/src/runtime/core/init/getEngineInstance.ts

```diff
@@ -0,0 +1,35 @@
+import { ClientEngineType, getClientEngineType } from '@prisma/internals'
+
+import { BinaryEngine, DataProxyEngine, EngineConfig, LibraryEngine } from '../engines'
+import { PrismaClientValidationError } from '../errors/PrismaClientValidationError'
+import { resolveDatasourceUrl } from './resolveDatasourceUrl'
+
+export function getEngineInstance(_engineConfig: EngineConfig) {
+  let url: string | undefined
+
+  try {
+    url = resolveDatasourceUrl({
+      inlineDatasources: _engineConfig.inlineDatasources,
+      overrideDatasources: _engineConfig.overrideDatasources,
+      env: { ..._engineConfig.env, ...process.env },
+      clientVersion: _engineConfig.clientVersion,
+    })
+  } catch {
+    // the error does not matter, but that means we don't have a valid url which
+    // means we can't use the DataProxyEngine and will default to LibraryEngine
```

Comment:
I'm not sure I get it 🤔 
Why not error then?

Reply:
It it errors, it means we have no URL and we default to `LibraryEngine`, to use the Data Proxy you need to provide the URL.

Reply:
Also a bit more context on this, we have tests that assert that errors are never thrown in the constructor but only can happen on the first query, so it would be incorrect to error on init. Maybe that answers the question. If this is ever changed, the test will let us know.